### PR TITLE
Trasnport: use config properties instead user properties.

### PR DESCRIPTION
**Pull request by cstamas**
This setting should be possible to be set via MAVEN_OPTS (system wide) or in .mvn/maven.config etc, Currently it is possible only via `-Dxxx` AFTER mvn....

### CODE REVIEW

File: maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java [hunk 1]
```diff
@@ -261,8 +261,7 @@ else if ( request.isUpdateSnapshots() )
         }
         session.setAuthenticationSelector( authSelector );
 
-        String transport = request.getUserProperties()
-                .getProperty( MAVEN_RESOLVER_TRANSPORT_KEY, MAVEN_RESOLVER_TRANSPORT_WAGON );
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:265]: So this is a merged, final version of the properties and the actual source is irrelevant?

Comment (cstamas) [reply maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:265]: yup, this is "config" properties (merged plus added things). I think I understand from where "request.systemProperties" are sourced, but from where are "request.userProperties" sourced?

Comment (michael-o) [reply maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:265]: Look into Maven CLI, that is the source for it. Commons CLI provides it and it is passed down the line.

Comment (slawekjaranowski) [reply maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:265]: `request.*Properties` is populated in 
https://github.com/apache/maven/blob/master/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

**Review by michael-o — CHANGES_REQUESTED** [review]
Please create a JIRA issue, this needs to go into 3.9.0 and master.

**Comment by cstamas** [comment]
Do we want to address build consumer feature as well? As it suffers from very same issue

**Comment by michael-o** [comment]
No, at least not for now. It is a user feature  therefore a user property

**Comment by cstamas** [comment]
Superseded by https://github.com/apache/maven/pull/739

**Pull request closed without merging**